FINERACT-2181: Fix github actions not failing test runs on failed tests

magyari-adam opened the pull request:
- added validation for cumulative multidisburse loan applications for interest method
- test fixes

## Description

Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Write the commit message as per https://github.com/apache/fineract/#pull-requests
- [x] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.
- [x] Create/update unit or integration tests for verifying the changes made.
- [x] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.
- [x] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes
- [x] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

adamsaghy commented:
@magyari-adam Can you please sum this PR up to me?

adamsaghy commented on fineract-e2e-tests-runner/src/test/resources/features/LoanDownPayment.feature:
```diff
@@ -1108,14 +1108,14 @@ Feature: Loan DownPayment
-      | 3  | 0    | 15 February 2022 |                  | 1250.0          | 250.0         | 0.0      | 0.0  | 0.0       | 250.0 | 0.0   | 0.0        | 0.0   | 250.0       |
-      | 4  | 28   | 01 March 2022    |                  | 750.0           | 500.0         | 0.0      | 0.0  | 0.0       | 500.0 | 0.0   | 0.0        | 0.0   | 500.0       |
-      | 5  | 31   | 01 April 2022    |                  | 0.0             | 750.0         | 0.0      | 0.0  | 0.0       | 750.0 | 0.0   | 0.0        | 0.0   | 750.0       |
+      | 3  | 0    | 15 February 2022 | 15 February 2022 | 1250.0          | 250.0         | 0.0      | 0.0  | 0.0       | 250.0 | 250.0 | 0.0        | 0.0   | 0.0         |
```
Looks not correct to me.. How could we pay a later period (3) before an earlier, still not paid period (2)?

magyari-adam replied:
There was a misunderstanding on my part about where should a downpayment allocate, fixed it.

adamsaghy reviewed (changes requested):
See my comment

adamsaghy commented on integration-tests/src/test/java/org/apache/fineract/integrationtests/LoanTransactionAccrualActivityPostingTest.java:
```diff
@@ -1203,21 +1203,21 @@ public void testAccrualActivityPostingForMultiDisburseLoan() {
             verifyTransactions(loanId.get(),
-                    transaction(500.0, "Disbursement", disbursementDay2, 479.16, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, false), //
-                    transaction(650.0, "Repayment", repaymentDate1, 0.0, 520.84, 59.16, 40.0, 30.0, 0.0, 0.0, false), //
+                    transaction(500.0, "Disbursement", disbursementDay2, 479.18, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, false), //
```
@magyari-adam This changes are worries me.... :/

adamsaghy commented on fineract-e2e-tests-runner/src/test/resources/features/LoanDownPayment.feature:
```diff
@@ -1111,8 +1111,8 @@ Feature: Loan DownPayment
       | 3  | 0    | 15 February 2022 |                  | 1250.0          | 250.0         | 0.0      | 0.0  | 0.0       | 250.0 | 0.0   | 0.0        | 0.0   | 250.0       |
-      | 4  | 28   | 01 March 2022    |                  | 750.0           | 500.0         | 0.0      | 0.0  | 0.0       | 500.0 | 0.0   | 0.0        | 0.0   | 500.0       |
-      | 5  | 31   | 01 April 2022    |                  | 0.0             | 750.0         | 0.0      | 0.0  | 0.0       | 750.0 | 0.0   | 0.0        | 0.0   | 750.0       |
+      | 4  | 28   | 01 March 2022    |                  | 625.0           | 625.0         | 0.0      | 0.0  | 0.0       | 625.0 | 0.0   | 0.0        | 0.0   | 625.0       |
```
I am afraid something here is still incorrect. The EMI should be 250 for every installment. On cumulative loans it doesnt really matter on which installment the disbursement occurred, it is divided across every period anyway.

magyari-adam committed:
FINERACT-2181: Fix github actions not failing test runs on failed tests

- added validation for cumulative multidisburse loan applications for interest method
- test fixes
